### PR TITLE
Use io._vector_fields to get a field shape. Fixes #3191

### LINKED ISSUE #3191: Field detection hardcodes vector fields
As discovered by @Xarthisius while working with @jcoughlin11 , the current field detection method hardcodes the names of possible vector fields. This leads to not detecting aliases for common names like `positions`.

It should be possible to check the vector field list, since that ought to be available at runtime through the `io` object.

### PR DESCRIPTION
**Pull request by Xarthisius**
I need to check if that doesn't introduce any unnecessary overhead, but on the bright side it removes a lot of hardcoded information.

### CODE REVIEW

**Review by matthewturk — APPROVED** [review]
definite improvement

File: yt/fields/field_detector.py [hunk 1]
```diff
@@ -146,24 +147,17 @@ def __missing__(self, item):
                     self[item] = vv.ravel()
                 return self[item]
         elif finfo is not None and finfo.sampling_type == "particle":
-            if (
-                "particle_position" in (item, item[1])
```
Comment (munkm) [inline yt/fields/field_detector.py:150]: Is it ok that we're not checking `particle_position`, `particle_velocity`, or `particle_magnetic_field` anymore with these updates (unless I'm missing something?)?

**Review by munkm — APPROVED** [review]
Generally this looks good! I have one question and am happy to merge once it's answered!

File: yt/fields/field_detector.py (hunk 1 above)
Comment (Xarthisius) [reply yt/fields/field_detector.py:150]: tbh I don't know. However, no test fails and I couldn't identify the place where it was used (if it was used).

Comment (Xarthisius) [reply yt/fields/field_detector.py:150]: I digged a little. Normally those fields are aliases to some frontend specifc fields (like `particle_position` -> `Coordinates` for Gadget etc.). They'd only play a role if a frontend had a native field named like that, but then it should be defining it in its specific `IOHandler._vector_fields`.

**Comment by jcoughlin11** [comment]
@Xarthisius @matthewturk @munkm I'm just wondering if this is ready to be merged? I ask because it would help a ton for my work on the qmc frontend. No rush, though!

**Comment by Xarthisius** [comment]
As far as I'm concerned this is ready to go.

**Pull request merged**